tests/src/test/scala/com/nvidia/spark/rapids/AdaptiveQueryExecSuite.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.execution.{PartialReducerPartitionSpec, SparkPlan}
+import org.apache.spark.sql.execution.adaptive.{AdaptiveSparkPlanExec, AdaptiveSparkPlanHelper}
+import org.apache.spark.sql.execution.exchange.Exchange
+import org.apache.spark.sql.functions.when
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.rapids.execution.{GpuCustomShuffleReaderExec, GpuShuffledHashJoinBase}
+
+class AdaptiveQueryExecSuite
+    extends SparkQueryCompareTestSuite
+        with AdaptiveSparkPlanHelper {
+
+  private def runAdaptiveAndVerifyResult(
+      spark: SparkSession, query: String): (SparkPlan, SparkPlan) = {
+
+    val dfAdaptive = spark.sql(query)
+    val planBefore = dfAdaptive.queryExecution.executedPlan
+    assert(planBefore.toString.startsWith("AdaptiveSparkPlan isFinalPlan=false"))
```

Comment:
I really dislike `toString` being used for test unless `toString` is what we are testing.  Is there a cleaner way to do this?

Reply:
This is copied directly from the test in Spark. Unfortunately `isPrivatePlan` is a private field in `AdaptiveSparkPlanExec` so this is how they check for final plans in their tests. There is no other way to access it that I am aware of (other than reflection).

Reply:
OK, then could you add a comment explaining why we are using `toString`?